### PR TITLE
modules/ignition,platforms/*: Enable install behind HTTP proxy

### PR DESCRIPTION
**Pull request by lander2k2**
This addition will allow a cluster to be installed behind an HTTP proxy
on all platforms.

### CODE REVIEW

File: platforms/azure/main.tf [hunk 1]
```diff
@@ -184,6 +194,8 @@ module "ignition_workers" {
   kubelet_debug_config    = "${var.tectonic_kubelet_debug_config}"
   kubelet_node_label      = "node-role.kubernetes.io/node"
   kubelet_node_taints     = ""
+  no_proxy                = "${var.tectonic_no_proxy}"
+  tectonic_vanilla_k8s    = "${var.tectonic_vanilla_k8s}"
```
Comment (enxebre) [inline platforms/azure/main.tf]: tectonic_vanilla_k8s has been removed, need rebase

Comment (lander2k2) [reply platforms/azure/main.tf]: Ugh.  Sorry 'bout that.  I did see it had been removed.  Mistakenly left in during merge conflict resolution.

**Comment by cpanato** [comment]
@lander2k2  @enxebre  lets wait for this https://github.com/coreos/tectonic-installer/pull/2662 to be merged and then we rebase this PR as well

**Comment by brianredbeard** [comment]
Does this actually need changes to 42 files?  If that's the case what's the argument for avoiding some type of inheritance to keep it [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)?

**Comment by lander2k2** [comment]
@brianredbeard In order to implement across all platforms, it does indeed need changes to 42 files.  8 of those files are just docs and examples, but nonetheless, there is considerable repetition.  The ignition module has improved the DRYness of these kinds of changes but deeper changes to the installer are probably a little beyond the scope of this PR.  We're definitely stretching the capabilities of terraform here.

**Comment by sym3tri** [comment]
DRYness issues are planned to be resolved in forthcoming PRs.

**Comment by brianredbeard** [comment]
@lander2k2 As this is referencing an internal resource which is not managed by the installing administrator, this should also allow for the specification of an internal certificate authority (common with the use of BlueCoat and similar products).

**Comment by lander2k2** [comment]
@brianredbeard I haven't encountered the internal CA requirement related to the proxy at Ford or T-Mobile.  I bet you're right about needing to do it in the future.  Hopefully, we'll be on track 2 by that time and will have a cleaner mechanism to meet that requirement.

**Comment by kalmog** [comment]
ok to test

**Comment by kalmog** [comment]
ok to test

**Pull request merged**